.github/workflows/benchmark_trigger.yml

```diff
@@ -32,39 +32,47 @@ jobs:
   # constraint on it; otherwise the irrelevant labeling events will cancel the
   # events that meet the preconditions. Even with cancel-in-progress = false,
   # the queued events will still be cancelled.
-  check:
+  precondition:
     runs-on: ubuntu-20.04
     outputs:
-      should-rerun: ${{ steps.precondition.outputs.should-rerun }}
+      found-label: ${{ steps.precondition.outputs.found-label }}
+      triggered-at: ${{ steps.precondition.outputs.triggered-at }}
     env:
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - name: "Checking labels"
         id: precondition
         run: |
-          echo "should-rerun=$(jq --raw-output \
+          echo "found-label=$(jq \
             --arg label_prefix ${BENCHMARK_LABEL_PREFIX} \
             '.label.name | startswith($label_prefix)' \
             ${GITHUB_EVENT_PATH})" >> "${GITHUB_OUTPUT}"
```

Comment:
So this reads from the GitHub event that triggered this workflow. Doesn't this mean that if I add 2 labels (`benchmark:android` and `benchmark:cuda` for instance) that GitHub will create two events. The first event will only have the `benchmark:android` label. But working through the logic here, this doesn't matter because we actually trigger a different run that *itself* checks the labels, right?

Reply:
Correct, the CI workflow will read labels again and it should get all labels. There might still be a race condition if CI workflow is restarted too fast while Github is still populating the labels? But I haven't observed that yet.